### ISSUE TITLE
Test suite is broken by base16-bytestring >= 1.0

### ISSUE THREAD

**Issue by sternenseemann** [serokell/haskell-crypto#25]
```
[4 of 7] Compiling Test.Crypto.Pwhash ( test/Test/Crypto/Pwhash.hs, dist/build/test/test-tmp/Test/Crypto/Pwhash.o )

test/Test/Crypto/Pwhash.hs:37:21: error:
    • Couldn't match expected type ‘(ba0, a2)’
                  with actual type ‘Either String ByteString’
    • In the expression: B16.decode hash
      In a pattern binding: (hash', "") = B16.decode hash
      In the expression:
        do let (hash', "") = B16.decode hash
           let (passwd', "") = B16.decode passwd
           let (salt', "") = B16.decode salt
           let Just salt'N = sizedByteArray (BS.take 16 salt')
           ....
    • Relevant bindings include
        hash' :: ba0 (bound at test/Test/Crypto/Pwhash.hs:37:8)
   |
37 |   let (hash', "") = B16.decode hash
   |                     ^^^^^^^^^^^^^^^

test/Test/Crypto/Pwhash.hs:38:23: error:
    • Couldn't match expected type ‘(passwd0, a1)’
                  with actual type ‘Either String ByteString’
    • In the expression: B16.decode passwd
      In a pattern binding: (passwd', "") = B16.decode passwd
      In the expression:
        do let (hash', "") = B16.decode hash
           let (passwd', "") = B16.decode passwd
           let (salt', "") = B16.decode salt
           let Just salt'N = sizedByteArray (BS.take 16 salt')
           ....
    • Relevant bindings include
        passwd' :: passwd0 (bound at test/Test/Crypto/Pwhash.hs:38:8)
   |
38 |   let (passwd', "") = B16.decode passwd
   |                       ^^^^^^^^^^^^^^^^^

test/Test/Crypto/Pwhash.hs:39:21: error:
    • Couldn't match expected type ‘(ByteString, a0)’
                  with actual type ‘Either String ByteString’
    • In the expression: B16.decode salt
      In a pattern binding: (salt', "") = B16.decode salt
      In the expression:
        do let (hash', "") = B16.decode hash
           let (passwd', "") = B16.decode passwd
           let (salt', "") = B16.decode salt
           let Just salt'N = sizedByteArray (BS.take 16 salt')
           ....
   |
39 |   let (salt', "") = B16.decode salt
```

**Comment by kirelagin** [comment]
Oh, thanks for reporting (sorry, apparently I had notifications off for this repo :/). Looks like I forgot to specify version bounds for some of the test dependencies.

**Comment by kirelagin** [comment]
Fixed in aa2eb7d0015d98b59549a1e9e3ad2f7b8c26344e.
